sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/tests/ScaleHostEndToEndTests.cs

```diff
@@ -0,0 +1,183 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Azure.Messaging.ServiceBus.Tests;
+using Microsoft.Azure.WebJobs.Host.EndToEndTests;
+using Microsoft.Azure.WebJobs.Host.Scale;
+using Microsoft.Azure.WebJobs.Host.TestCommon;
+using Microsoft.Extensions.Azure;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json.Linq;
+using NUnit.Framework;
+
+namespace Microsoft.Azure.WebJobs.Extensions.ServiceBus.Tests
+{
+    internal class ScaleHostEndToEndTests : WebJobsServiceBusTestBase
+    {
+        private const string Function1Name = "Function1";
+        private const string Function2Name = "Function2";
+
+        public ScaleHostEndToEndTests() : base(isSession: false)
+        {
+        }
+
+        [Test]
+        [TestCase(false)]
+        [TestCase(true)]
+        public async Task ScaleHostEndToEndTest(bool tbsEnabled)
+        {
+            string hostJson =
+            @"{
+                ""azureWebJobs"" : {
+                    ""extensions"": {
+                        ""serviceBus"": {
+                            ""maxConcurrentCalls"" :  1
+                        }
+                    }
+                }   
+            }";
+
+            // Function1Name uses connection string
+            // Function2Name uses AzureComponentFactory - simulating managed identity scenario in ScaleController
+            string triggers = $@"{{
+""triggers"": [
+    {{
+        ""name"": ""myQueueItem"",
+        ""type"": ""serviceBusTrigger"",
+        ""direction"": ""in"",
+        ""queueName"": ""{FirstQueueScope.QueueName}"",
+        ""functionName"": ""{Function1Name}""
+    }},
+    {{
+        ""name"": ""myQueueItem"",
+        ""type"": ""serviceBusTrigger"",
+        ""direction"": ""in"",
+        ""queueName"": ""{SecondQueueScope.QueueName}"",
+        ""connection"": ""ServiceBusConnection2"",
+        ""functionName"": ""{Function2Name}""
+    }}
+ ]}}";
+
+            IHost host = new HostBuilder().ConfigureServices(services => services.AddAzureClientsCore()).Build();
+            AzureComponentFactory defaultAzureComponentFactory = host.Services.GetService<AzureComponentFactory>();
+            AzureComponentFactoryWrapper factoryWrapper = new AzureComponentFactoryWrapper(defaultAzureComponentFactory, ServiceBusTestEnvironment.Instance.Credential);
```

Comment:
Is the only reason we are creating the wrapper type to be able to pass in the environment credential? If so, you can instead follow the pattern here - https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/tests/WebJobsServiceBusTestBase.cs#L143-L148

Reply:
We do have all the properties (clientId, clientSecret, tenantId) as user does not specify them. ScaleController/Antaers can make a special call to get the managed identity token credentials. Then we can pass them to the extension.

Reply:
Please add test to the name, e.g. TestComponentFactory.